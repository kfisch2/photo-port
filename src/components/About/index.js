import coverImage from "../../assets/cover/cover-image.jpg";
import React from "react";

function About() {
  return (
    <section className="my-5">
      <header id="about">
        <h1>About</h1>
      </header>
      <img src={coverImage} className="my-2" style={{ width: '100%'}}/>
    </section>
  );
}

export default About;
