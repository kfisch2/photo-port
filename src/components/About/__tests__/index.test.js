import React from "react";

import { render, cleanup } from "@testing-library/react";

import "@testing-library/jest-dom/extend-expect";

import About from "..";

// after each test, we won't have leftover memory data
afterEach(cleanup);

describe("About component", () => {
  // renders About test
  // First test
  // 'it' first arg = string of what's being tested
  //      second arg =  cb that runs the test
  it("renders", () => {
    render(<About />);
  });

  // Second test
  it("matches snapshot DOM node structure", () => {
    // render About
    const { asFragment } = render(<About />);
    expect(asFragment()).toMatchSnapshot();
  });
});
