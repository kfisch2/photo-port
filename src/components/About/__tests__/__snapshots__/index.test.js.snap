// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <section
    class="my-5"
  >
    <header
      id="about"
    >
      <h1>
        About
      </h1>
    </header>
    <img
      class="my-2"
      src="cover-image.jpg"
      style="width: 100%;"
    />
  </section>
</DocumentFragment>
`;
